### PR TITLE
Fix panic error when unwrapping root_uri without workspace

### LINKED ISSUE #36: Extension setting page for typst has no setting
I use code oss 1.76 version in linux.  Extension setting page for typst has no setting.
image

### LINKED ISSUE #41: Doesn't create a PDF
Perhaps a dumb question, but I installed the VSCode extension (on Windows 10), created a .typ file, entered some random text, saved it. Nothing happens. No pdf is created, nothing in Output window. VSCode is showing "Typst" for document and extension settings for PDF creation are set "onSave".
Am I missing something obvious? Also, perhaps add a paragraph to readme on what to expect (i.e. if configured properly, a NAME.pdf should appear alongside original text).

### PR DESCRIPTION
**Pull request by Bramas**
Fix the following error when opening a singe file (with no folder in the workspace)
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/main.rs:26:29
```

fixes #41 
fixes #36

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -33,10 +33,14 @@ struct Backend {
...
-        *world = Some(SystemWorld::new(
-            params.root_uri.unwrap().to_file_path().unwrap(),
-        ));
-
+        // Check if a folder is opened, if yes, use it as the root path
+        let root_path = if let Some(root) = params.root_uri {
+            root.to_file_path().unwrap()
+        } else {
+            PathBuf::new()
+        };
```
Comment (beeb) [inline src/main.rs:41]: Personal preference but I would rewrite this as a match statement:

```rust
let root_path = match params.root_uri {
    Some(root) => root.to_file_path().unwrap(),
    None => PathBuf::new()
};
```

or even to:

```rust
let root_path = params.root_uri.map(|root| root.to_file_path().unwrap()).unwrap_or_else(|| PathBuf::new());
```

I think either of those look more idiomatic.

Comment (Bramas) [reply src/main.rs:41]: Yes definitly, I like your first choice. I am quiet new to Rust so thanks for the comment :)

**Comment by beeb** [comment]
> What happens with an empty PathBuf? Was it tested with both opening a workspace and opening a single file?

I just tested and the fix works for me

**Comment by nvarner** [comment]
@Bramas this looks good, so planning to merge. Thank you for the fix! You mentioned not being able to compile, though. Would you mind opening an issue with the errors you're getting?

**Comment by nvarner** [comment]
@Bramas the formatting check also failed. You should be able to see the exact errors in the failed check (Rust prefers trailing commas, and there was some stray whitespace). However, you can also run `cargo fmt` to have it fix it for you.

**Comment by Bramas** [comment]
Thanks for your help.
Everything compiles now (so I don't know what it was)

**Pull request merged**